src/bun.js/node/node_os.zig

```diff
@@ -294,10 +294,193 @@ pub const Os = struct {
+                    std.os.AF.INET6 => netmaskToCIDRSuffix(@bitCast(u128, netmask.in6.sa.addr)),
+                    else => null
+                };
+
+                // Format the address and then, if valid, the CIDR suffix; both
+                //  the address and cidr values can be slices into this same buffer
+                // e.g. addr_str = "192.168.88.254", cidr_str = "192.168.88.254/24"
+                var buf: [64]u8 = undefined;
+                const addr_str = formatAddress(addr, &buf) catch unreachable;
+                var cidr = JSC.JSValue.null;
+                if (maybe_suffix) |suffix| {
+                    //NOTE addr_str might not start at buf[0] due to slicing in formatAddress
+                    const start = @ptrToInt(addr_str.ptr) - @ptrToInt(&buf[0]);
+                    // Start writing the suffix immediately after the address
+                    const suffix_str = std.fmt.bufPrint(buf[start + addr_str.len..], "/{}", .{ suffix }) catch unreachable;
+                    // The full cidr value is the address + the suffix
+                    const cidr_str = buf[start..start + addr_str.len + suffix_str.len];
+                    cidr = JSC.ZigString.init(cidr_str).withEncoding().toValueGC(globalThis);
+                }
+
+                interface.put(globalThis, JSC.ZigString.static("address"),
+                                          JSC.ZigString.init(addr_str).withEncoding().toValueGC(globalThis));
+                interface.put(globalThis, JSC.ZigString.static("cidr"), cidr);
+            }
+
+            // netmask <string> The IPv4 or IPv6 network mask
+            {
+                var buf: [64]u8 = undefined;
+                const str = formatAddress(netmask, &buf) catch unreachable;
+                interface.put(globalThis, JSC.ZigString.static("netmask"),
+                                          JSC.ZigString.init(str).withEncoding().toValueGC(globalThis));
+            }
+
+            // family <string> Either IPv4 or IPv6
+            interface.put(globalThis, JSC.ZigString.static("family"),
+                (switch (addr.any.family) {
+                    std.os.AF.INET => JSC.ZigString.static("IPv4"),
+                    std.os.AF.INET6 => JSC.ZigString.static("IPv6"),
+                    else => JSC.ZigString.static("unknown"),
+                }).toValue(globalThis)
+            );
+
+            // mac <string> The MAC address of the network interface
+            {
+                // We need to search for the link-layer interface whose name matches this one
+                var ll_it = interface_start;
+                const maybe_ll_addr = while (ll_it) |ll_iface| : (ll_it = ll_iface.ifa_next) {
+                    if (helpers.skip(ll_iface) or !helpers.isLinkLayer(ll_iface)) continue;
+
+                    const ll_name = std.mem.sliceTo(ll_iface.ifa_name, 0);
+                    if (!std.mem.startsWith(u8, ll_name, interface_name)) continue;
```

Comment:
```suggestion
                    if (!strings.hasPrefix(ll_name, interface_name)) continue;
```

we should also make our startsWith wrapper use our faster string equality checker

<img width="532" alt="image" src="https://user-images.githubusercontent.com/709451/221301341-476672ca-b7b8-404f-9cc8-4c83b2fb2229.png">